resources/todomvc/architecture-examples/preact/src/app/main.js

```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line no-unused-vars
+import { h } from "preact";
+
+import TodoItem from "./item";
+import { FILTERS } from "./utils";
+
+export default function TodoMain({ onChange, onToggle, onRemove, onSave, todos, route }) {
+    const visibleTodos = todos.filter(FILTERS[route]);
+    const activeTodoCount = todos.filter(FILTERS["active"]).length;
+
+    return (
+        <section class="main">
+            <input class="toggle-all" type="checkbox" onChange={onChange} checked={activeTodoCount === 0} />
```

Comment:
This doesn't seem to be displayed, or am I missing something?

Reply:
@julienw - regarding the use of Date.now() - I tried previously two things:
**the above mentioned forceUpdate solution from the doc** 
using useReducer comes with a cost and performance degraded in preact

**using something simpler like val +1**
this worked when running the workload by itself, but when running it in a test in speedometer it broke.

My assumption is that Date.now() produces a more unique number and ensures that useState doesn't get called with the same value for some reason and therefore prevents an update.

I could be totally off though.

Reply:
I think you replied to the wrong comment :)

>  using useReducer comes with a cost and performance degraded in preact

🤔 this is strange but I don't know the hooks enough (nor preact) to understand why this might be.

> using something simpler like val +1

This should work if you use the function style of setting the state: `setValue(val => val+1)` (in this case React will call the function with the value as a parameter) instead of `setValue(value + 1)` where `value` comes from `useState` (in this case we may have an outdated value because of some race).

But no need to spend too much time on this, this is some details. This could also be looked at in follow-ups, we don't need to make it perfect in this PR.